### PR TITLE
post via burn

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: contracts/v4/Tipping_v4.aes [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE
+// OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+// PERFORMANCE OF THIS SOFTWARE.
+
+
+// THIS IS NOT SECURITY AUDITED
+// DO NEVER USE THIS WITHOUT SECURITY AUDIT FIRST
+
+@compiler >= 4.2
+//@compiler >= 6
```
Comment (kenodressel) [inline contracts/v4/Tipping_v4.aes]: why comment it out? only works with iris?

File: contracts/v4/Tipping_v4.aes [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  datatype tip = PostViaBurn(tip_meta, list(string), TokenContract, token_amount)
+
+  record tip_meta = { sender    : address
+                    , title     : string
+                    , timestamp : int }
+
+  record state = { tips    : map(tip_id, tip)
+                 , version : string }
+
+  datatype event = PostViaBurnReceived(tip_id, address, string)
```
Comment (kenodressel) [inline contracts/v4/Tipping_v4.aes]: maybe add the token here?

**Review by kenodressel — APPROVED** [review]

File: contracts/v4/Tipping_v4.aes (hunk 1 above)
Comment (thepiwo) [reply contracts/v4/Tipping_v4.aes]: aeproject is not updated yet

File: contracts/v4/Tipping_v4.aes (hunk 2 above)
Comment (thepiwo) [reply contracts/v4/Tipping_v4.aes]: good idea

**Comment by mradkov** [comment]
Can you revisit this now after aeproject was updated with Iris support? Or maybe rebase on top of #30.

**Comment by thepiwo** [comment]
@mradkov yes, thats the plan

**Review by kenodressel — APPROVED** [review]
lgtm

**Pull request merged**